File: 10. Subprogram/F. Teman Dekat/f_teman_dekat.cpp
#include <iostream>

int power(int base, int exp){
  int res = base;
  while (exp > 1){
    res *= base;
    exp--;
  }
  return res;
}

int main(){
  int N, D, i, j, res, min, max;
  std::cin >> N >> D;
  int T[N][2];
  for (i = 0; i < N; i++)
    for (j = 0; j < 2; j++)
      std::cin >> T[i][j];

  for (i = 0; i < (N - 1); i++){
    for (j = i + 1; j < N; j++){
      res = power(abs(T[j][0] - T[i][0]), D) + power(abs(T[j][1] - T[i][1]), D);
      if (i == 0 && j == 1){
        min = max = res;
      }
      else{
        if (res < min) min = res;
        if (res > max) max = res;
      }
    }
  }
  std::cout << min << " " << max << std::endl;
  return 0;
}

File: 10. Subprogram/F. Teman Dekat/f_teman_dekat.java
import java.util.Scanner;

class f_teman_dekat {
    public static int abs(int num) {
        if (num < 0) {
            return -num;
        }
        return num;
    }

    public static int power(int base, int exp) {
        if (exp == 0) {
            return 1;
        }
        return base * power(base, exp - 1);
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int min = 0, max = 0;
        int N = scan.nextInt();
        int D = scan.nextInt();
        int[][] T = new int[N][2];
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < 2; j++) {
                T[i][j] = scan.nextInt();
            }
        }
        scan.close();
        for (int i = 0; i < (N - 1); i++) {
            for (int j = i + 1; j < N; j++) {
                int res = power(abs(T[i][0] - T[j][0]), D) + power(abs(T[i][1] - T[j][1]), D);
                if (i == 0 && j == 1) {
                    min = res;
                    max = res;
                } else {
                    if (res < min) {
                        min = res;
                    }
                    if (res > max) {
                        max = res;
                    }
                }
            }
        }
        System.out.println(min + " " + max);
    }
}